blocks/blocks.go

```diff
@@ -96,28 +118,19 @@ func (bh *BlockHandler) HandleBlockData(ctx context.Context, data []byte) error
 
 	// check if known
 	if _, err := bh.mesh.GetBlock(blk.ID()); err == nil {
-		logger.Info("we already know this block")
+		logger.Info("received known block")
 		return nil
 	}
 	logger.With().Info("got new block", blk.Fields()...)
 
 	if err := bh.blockSyntacticValidation(ctx, &blk); err != nil {
-		logger.With().Error("failed to validate block", log.Err(err))
-		return fmt.Errorf("failed to validate block %v", err)
+		return fmt.Errorf("failed to validate block %w", err)
 	}
 
 	saveMetrics(blk)
 
 	if err := bh.mesh.AddBlockWithTxs(ctx, &blk); err != nil {
-		logger.With().Error("failed to add block to database", log.Err(err))
-		// we return nil here so that the block will still be propagated
-		return nil
-	}
-
-	if !blk.Layer().After(bh.mesh.ProcessedLayer()) { //|| blk.Layer() == bh.mesh.getValidatingLayer() {
-		logger.With().Warning("block is late",
-			log.FieldNamed("processed_layer", bh.mesh.ProcessedLayer()),
-			log.FieldNamed("miner_id", blk.MinerID()))
+		return fmt.Errorf("persisting block %s: %w", blk.ID(), err)
```

Comment:
adding block to mesh is not exactly the same as persisting...

Reply:
right, need to change, i checked why it can fail, and wrote a message based on that